react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  useEffect(() => {
+    async function loadItems() {
+      try {
+        const items = await readItems();
+        setItems(items);
+      }
+      catch (error){
+        setError(error)
+      }
+    }
+     setIsLoading(false);
```

Comment:
This doesn't do what you want. It sets isLoading to false _before_ the loading actually completes. You shouldn't set it to false until after the items are successfully loaded.